tests/chisel.spec.ts

```diff
@@ -0,0 +1,129 @@
+        });
+
+        it("return undefined when chisel URL is empty", async () => {
+            const serviceName = "chisel-service-name";
+            const  envPath = join(chiselTestDataPath, ".env-without-chisel");            
+            await writeProperties(tempFilePath, { "CHISEL_URL": ""});
+            expect(await checkAndCreateChiselTask(envPath, serviceName)).to.be.undefined;
+        });
+
+        it("return undefined when .env file does not exists", async () => {
+            const serviceName = "chisel-service-name";
+            const  envPath = join(chiselTestDataPath, ".env-not-exists");
+            expect(await checkAndCreateChiselTask(envPath, serviceName)).to.be.undefined;
+        });
+    });
+
+    describe("deleteChiselParamsFromFile", () => {
+        const expectedProperties = {
+            "VCAP_SERVICES": VCAP_SERVICE_WITH_ALL_CHARS
+        };
+        it("return true when delete chisel params from the .env file", async () => {
+            await writeProperties(tempFilePath, envWithChisel);
+            expect(await deleteChiselParamsFromFile(tempFilePath)).to.be.true;
+            const actualProperties = PropertiesReader(tempFilePath);
+            expect(actualProperties.getAllProperties()).deep.equal(expectedProperties);
+          
+        });
+
+        it("return false when chisel params does not exists in .env file", async () => {
+            await writeProperties(tempFilePath, expectedProperties);
+            expect(await deleteChiselParamsFromFile(tempFilePath)).to.be.false;
+            const actualProperties = PropertiesReader(tempFilePath);
+            expect(actualProperties.getAllProperties()).deep.equal(expectedProperties);
+        });
+
+        it("return false when .env file does not exists ", async () => {
+            expect(await deleteChiselParamsFromFile(tempFilePath)).to.be.false;
+        });
+/*
+  TODO: Add unite test when a write to a file when it failed 
+        it.only("return false when write to .env file is failed", async () => {
+            const envWithChisel = {   
+                "TUNNEL_PARAM": "2020:127.0.0.1:2020",
+                "CHISEL_URL": "https://chise-server.cfapps.test.hana.ondemand.com",
+                "CHISEL_USER": "chisel-user",
+                "CHISEL_PASSWORD": "chisel-password",
+                "VCAP_SERVICES": "test"
+            };
+            await writeProperties(tempFilePath, envWithChisel);
+            const expObj = {
+                "VCAP_SERVICES": "test"
+            };
+            //stub(utils, "writeProperties").withArgs(tempFilePath, expObj).throwsException("mock failed to write to .env file");
+            chiselMock.expects("writeAsPropertiesFiles").withArgs(tempFilePath, expObj).throwsException("mock failed to write to .env file");
+            expect(await deleteChiselParamsFromFile(tempFilePath)).to.be.false;
+          
+        });
+*/
```

Comment:
?

Reply:
I wanted to add a unit test when the write failed but the mock didn't work 
I will remove it